Tests/Functional/Drivers/Queue/PhpAmqpLibDriverTest.php

```diff
@@ -0,0 +1,244 @@
+        $this->assertTrue($this->driver->isConnected());
+        $this->driver->destroy($this->createConsumer()->getName());
+        $this->assertFalse($this->driver->isConnected());
+    }
+
+     /**
+     * Tests that tries to declare a channel without having a connection
+     *
+     * @group channel-exception
+     */
+    public function testDeclareChannelWithoutConnnection()
+    {
+        $this->expectException(AMQPConnectionClosedException::class);
+        $this->expectExceptionMessage('Broken pipe or closed connection');
+        $this->assertTrue($this->driver->isConnected());
+        $this->driver->declareChannel();
+        $this->driver->destroy($this->createConsumer()->getName());
+        $this->assertFalse($this->driver->isConnected());
+        $this->driver->declareChannel();
+    }
+
+    /**
+     * @group format
+     */
+    public function testFormat()
+    {
+        $format = 'text/plain';
+        $this->driver->setFormat($format);
+        $this->assertEquals($format, $this->driver->getFormat());
+    }
+
+    /**
+     * @group amqp-connection
+     * @throws \AMQPConnectionException
+     */
+    public function testConnectWithoutData()
+    {
+        $this->driver->declareChannel();
+        $consumer = $this->createConsumer();
+        $this->driver->destroy($consumer->getName());
+        $this->driver->configure('', '', '', '');
+        $this->driver->connect(true);
+        $this->assertTrue($this->driver->isConnected());
+        $this->assertInstanceOf(AMQPStreamConnection::class, $this->driver->declareChannel()->getConnection());
+    }
+
+    /**
+     * @dataProvider getMessages
+     * @group amqp-consume-callback
+     */
+    public function testConsumeWaitNoBlock(MessageInterface $msg)
+    {
+        $this->prepareToConsume($msg);
+        $amqpMessage = null;
+
+        $callback = function($message) use (&$amqpMessage){
+            $amqpMessage = $message;
+            $queueMessage = new QueueMessage();
+            $queueMessage->setMessageId($message->getDeliveryTag());
+            $this->driver->ack($queueMessage);
+        };
+
+        $this->driver->consume($this->consumer->getName(), $this->queueName, $callback);
+        $this->driver->isConsuming();
+        $this->driver->waitNoBlock();
+
+        $this->assertInstanceOf(AMQPMessage::class, $amqpMessage);
+        $this->assertEquals($amqpMessage->delivery_info['routing_key'], $this->queueName);
+        $this->assertInstanceOf(AsyncQueueConsumer::class, $this->consumer);
+        $this->assertEquals($amqpMessage->delivery_info['consumer_tag'], $this->consumer->getName());
+        $this->assertContains('QueueMessage', $amqpMessage->getBody());
+    }
+
+    /**
+     * Prepare the data and class to consume a message
+     *
+     * @param QueueMessage $msg
+     * @return mixed
+     */
+    public function prepareToConsume(MessageInterface $msg)
+    {
+        $this->endpoint = $this->createMock(EndpointInterface::class);
```

Comment:
In this case it is not needed to mock and endpoint because we are testing the driver with the broker

Reply:
Ok, removed.

Reply:
You removed the endpoint but you missed to delete the
```
use use Smartbox\Integration\FrameworkBundle\Core\Endpoints\EndpointInterface;
```

Reply:
Removed.

Reply:
:+1: